### PR TITLE
meta: Add property getter methods

### PR DESCRIPTION
**Pull request by lmars**
This makes it easier to get object properties.

### CODE REVIEW

**Comment by orenyodfat** [comment]
Though this PR is not referring these files ...
error.go and encode.go missing documenation for exported functions and structs.

File: meta.go [hunk 1]
```diff
@@ -102,6 +102,59 @@ func (o *Object) Type() string {
...
+	l, ok := v.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("key %q has type %T, not []interface{}", key, v)
+	}
+	return l, nil
+}
+
+// Get returns the property with the given key.
+func (o *Object) Get(key string) (interface{}, error) {
+	v, rest, err := o.node.Resolve([]string{key})
```
Comment (orenyodfat) [inline meta.go]: What does "rest" means as a return value for Resolve .
I tried look up at https://github.com/ipfs/go-ipld-cbor/   but the function is not documented.

Comment (lmars) [inline meta.go]: `rest` represents the "rest" of the path which is left after finding a link whilst traversing the path.

If you have an object graph like this:

```
a = { "a": { "/": "<hash of b>" }
b = { "b": { "/": "<hash of c>" }
c = { "c": { "/": "<hash of d>" }
```

Then doing `a.Resolve([]string{"a", "b", "c"})` will return `<hash of b>` with rest being `[]string{"b", "c"}` meaning it found `a` but you need to resolve the rest starting from `b` once you have retrieved it.

We will improve this API in the future (I've already discussed future modifications of the IPLD CBOR lib), but for now this is what we have.

Comment (orenyodfat) [inline meta.go]: Ok. make sense.thanks.
i am suggesting to add the explanation above as part of the documentation of the function + TODO to improve the API.

Comment (lmars) [inline meta.go]: The `Resolve` function is from the cbor code though?

Comment (orenyodfat) [inline meta.go]: yea, but it is not well documented as your details explanation above.

Comment (lmars) [inline meta.go]: This function "returns the property with the given key" as documented, if you feel `Resolve` needs documenting then we need to do that on the cbor library.

Comment (orenyodfat) [inline meta.go]: I mean to add that as part of the Get function documanation (no need to document Resolve)

Comment (orenyodfat) [inline meta.go]: Ok . agree to leave the Get comment as is.

**Pull request merged**